### PR TITLE
feat: accept convertible types

### LINKED ISSUE #67: Custom types
Hi,

I'm reading some json data that contains a string with a timestamp that needs a custom `UnmarshalJSON` to unmarshal properly. I've crated a `type Timestamp time.Time` with an `UnmarshalJSON` method that works. Next I need to serialize this data to avro.

Naively just trying to use `avro.Marshal` on it unsurprisingly returns an `avro: mypkg.Timestamp is unsupported for Avro string` error.

Is there an easy way to get Avro to treat my custom `Timestamp` type as a string for a avro string field?

I can't seem to find anything similar to how `encoding/json` works where I just need to implement my own `MarshalJSON` function.

I'd like to avoid having to create the same struct again, just with string types for the timestamps, and copy over all the values to get it to serialize to Avro - its a fairly big struct.

### PR DESCRIPTION
**Pull request by TuSKan**
Fixes #67

### CODE REVIEW

File: codec_fixed.go [hunk 1]
```diff
@@ -26,9 +26,9 @@ func createDecoderOfFixed(schema Schema, typ reflect2.Type) ValDecoder {
 			break
 		}
 		switch {
-		case typ.RType() == durRType && ls.Type() == Duration:
+		case typ.Type1().ConvertibleTo(durType) && ls.Type() == Duration:
```
Comment (nrwiersma) [inline codec_fixed.go:29]: It would be more efficient to do the `typ.Type1()` once and reuse it. It is fairly expensive. This goes for all switch checks.

**Review by nrwiersma — CHANGES_REQUESTED** [review]

**Review by nrwiersma — APPROVED** [review]
LGTM 🎉

**Comment by nrwiersma** [comment]
Thanks for the contribution.

**Comment by TuSKan** [comment]
Thank you very much for accepting suggestions for improvements to your package. This will allow me to solve an important limitation in my application.
Many thanks also for quick response and iteration through the process.
There are some small improvements in the text of the README to better explain the new feature.
Let me know if you want another commit or prefer to do it yourself, or later.
Regards

**Pull request merged**